hugegraph-core/src/main/java/com/baidu/hugegraph/auth/StandardAuthManager.java

```diff
@@ -433,10 +450,234 @@ public RolePermission loginUser(String username, String password) {
+                project.adminGroupId(adminGroupId);
+            }
+
+            // Create project op group
+            if (project.opGroupId() == null) {
+                HugeGroup opGroup = new HugeGroup(project.opGroupName());
+                opGroup.creator(project.creator());
+                Id opGroupId = this.createGroup(opGroup);
+                project.opGroupId(opGroupId);
+            }
+
+            // Create project target to verify permission
+            HugeResource resource = new HugeResource(ResourceType.PROJECT,
+                                                     project.name(),
+                                                     null);
+            HugeTarget target = new HugeTarget(project.targetName(),
+                                               this.graph.name(),
+                                               "localhost:8080",
+                                               ImmutableList.of(resource));
+            target.creator(project.creator());
+            Id targetId = this.targets.add(target);
+            project.targetId(targetId);
+
+            Id adminGroupId = project.adminGroupId();
+            Id opGroupId = project.opGroupId();
+            HugeAccess adminGroupWriteAccess2Target = new HugeAccess(adminGroupId,
+                                                                     targetId,
+                                                                     HugePermission.WRITE);
+            adminGroupWriteAccess2Target.creator(project.creator());
+            HugeAccess adminGroupReadAccess2Target = new HugeAccess(adminGroupId,
+                                                                    targetId,
+                                                                    HugePermission.READ);
+            adminGroupReadAccess2Target.creator(project.creator());
+            HugeAccess opGroupReadAccess2Target = new HugeAccess(opGroupId,
+                                                                 targetId,
+                                                                 HugePermission.READ);
+            opGroupReadAccess2Target.creator(project.creator());
+            this.access.add(adminGroupWriteAccess2Target);
+            this.access.add(adminGroupReadAccess2Target);
+            this.access.add(opGroupReadAccess2Target);
+            return this.project.add(project);
+        });
+    }
+
+    @Override
+    public HugeProject deleteProject(Id id) {
+        return this.commit(() -> {
+            LockUtil.Locks locks = new LockUtil.Locks(this.graph.name());
+            try {
+                locks.lockWrites(LockUtil.PROJECT_UPDATE, id);
+
+                HugeProject oldProject = this.project.get(id);
+                E.checkArgumentNotNull(oldProject,
+                                       "The project with id '%s' " +
+                                       "can't be found", id);
+                // Check not graph bind this project
```

Comment:
improve "Check not graph bind this project"

Reply:
Check whether there are any graph binding this project, throw ForbiddenException, if it is